pennylane/circuit_graph.py

```diff
@@ -469,9 +470,59 @@ def get_depth(self):
                 self._operation_graph = self._graph.subgraph(
                     list(self._graph.nodes().index(node) for node in self.operations)
                 )
-                self._depth = rx.dag_longest_path_length(self._operation_graph) + 1
+                self._extend_graph(self._operation_graph)
+                self._depth = (
+                    rx.dag_longest_path_length(
+                        self._operation_graph, weight_fn=lambda _, __, w: self._weight_func(w)
+                    )
+                    + 1
+                )
         return self._depth
 
+    @staticmethod
+    def _weight_func(weight):
+        """If weight is a number, use it!"""
+        if isinstance(weight, (int, float)):
+            return weight
+        return 1
+
+    def _extend_graph(self, graph: rx.PyDiGraph) -> rx.PyDiGraph:
```

Comment:
How efficient (inefficient) this become for large circuits or circuits with custom operations that have large depths?

Reply:
Actually, @trbromley, found a nice solution. It should be efficient now since we are only extending it by 1 edge for a custom operation with any depth.